src/dscim/utils/rff.py

```diff
@@ -8,6 +8,196 @@
 import sys
 from numpy.testing import assert_allclose
 from datetime import datetime
+import fsspec
+import pyarrow
+import gurobipy as gp
+import time
+from scipy.sparse import coo_matrix
+from gurobipy import GRB
+
+
+## Solve the optimization problem
+def solve_optimization(ssp_df, rff_df):
+
```

Comment:
@davidrzhdu would be good to get some numpy docstring documentation for these new functions, describing their purpose, input args, and outputs. Let me know if you have any questions and thank you!